src/aws/flb_aws_credentials_sts.c

```diff
@@ -568,8 +576,11 @@ struct flb_aws_provider *flb_eks_provider_create(struct flb_config *config,
         return NULL;
     }
 
-    implementation->endpoint = flb_aws_endpoint("sts", region);
-    if (!implementation->endpoint) {
+     implementation->endpoint = removeProtocol(sts_endpoint, "https://");
```

Comment:
Move this line inside of the if block below; strstr behavior is undefined if any of the parameters are NULL, so this code might work for you on your compiler, but it might not work on others: https://stackoverflow.com/questions/19579574/what-is-behavior-of-null-parameters-to-strstr